### ISSUE TITLE
Question: how to display two views in the same page?

### ISSUE THREAD

**Issue by hainm** [molstar/molstar#1389]
hi, this question is for molstar in Jupyter notebook.
Basically I create two views, one have loaded structure and another is empty. However, displaying 2nd view will show the view from the the first view, just like two views are using the same container.

Do you have any why? Thanks!

Below is the JS code:
```Python
    async initializeDisplay() {
        this.setupContainer();

        var defaultSpec = DefaultPluginSpec();
        var spec: PluginUISpec = {
            ...defaultSpec,
            behaviors: [
            ...defaultSpec.behaviors,
            PluginSpec.Behavior(MolViewSpec)
            ]
        };
        this.plugin = await createPluginUI({
            target: this.container,
            render: (component, container) => {
                renderReact18(component, container);
            },
            spec: spec,
            onBeforeUIRender: async (ctx) => {
                // Implement the onBeforeUIRender logic here
            }
        });
        this._focused = false;
        await this.checkLeaderView();
    }

    setupContainer() {
        var container = document.createElement('div');
        container.id = 'container-' + Math.random().toString(36).substring(2, 11);
        container.style.width = '800px';
        container.style.height = '600px';
        this.el.appendChild(container);
        this.container = container;
    }
```

And here is the Python:
<img width="1425" alt="image" src="https://github.com/user-attachments/assets/f1ef14e7-8791-4804-b4b2-40a92a71c890" />

**Comment by dsehnal** [comment]
This seems like you are somehow sharing the same plugin context between the separate views. Would need more context from your implementation to provide more feedback.

**Comment by hainm** [comment]
Hi

The code is above. When a new view is created, the new plugin is created
too (if I understand correctly)


this.plugin = await createPluginUI({
            target: this.container,
            render: (component, container) => {
                renderReact18(component, container);
            },
            spec: spec,
            onBeforeUIRender: async (ctx) => {
                // Implement the onBeforeUIRender logic here
            }
        });

**Comment by dsehnal** [comment]
Yeah, but there is a lot more code around this as well. Is it possible for example that somehow the load trajectory command is sent to the newly created separate view as well?

**Comment by hainm** [comment]
> Yeah, but there is a lot more code around this as well. Is it possible for example that somehow the load trajectory command is sent to the newly created separate view as well?

I see. Please the the chunk of related code (and the whole file/folder) here: https://github.com/nglviewer/nglview/blob/53daa837bdd9da57943406c27a1a1e792a4efdf8/js/src/molstarview/widget.ts#L168-L178

**Comment by dsehnal** [comment]
In `WidgetBase` you have `_callbacks_before_loaded = []` as a class property.

```py
In [5]: class A:
   ...:     xs = []
   ...:

In [6]: a = A()

In [7]: a.xs.append(1)

In [8]: A().xs
Out[8]: [1]
```

could that be the cause of your issue?

**Comment by hainm** [comment]
> In `WidgetBase` you have `_callbacks_before_loaded = []` as a class property.
> 
> In [5]: class A:
>    ...:     xs = []
>    ...:
> 
> In [6]: a = A()
> 
> In [7]: a.xs.append(1)
> 
> In [8]: A().xs
> Out[8]: [1]
> 
> could that be the cause of your issue?

@dsehnal wow, wonderful. That's the issue. It happened during my code refactoring to support molstar. Thank you so much!.
